backend/app/api/endpoints/upload.py

```
@@ -524,3 +581,128 @@ async def upload_startup_files_batch(
+            except Exception as cleanup_error:
+                # Log cleanup error but don't mask original error
+                print(f"Cleanup failed: {cleanup_error}")
+
+        # Re-raise the original exception
+        if isinstance(e, HTTPException):
+            raise
+        else:
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail=f"Atomic batch upload failed: {str(e)}",
+            )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)